dronegen/tag.go

```diff
@@ -0,0 +1,425 @@
+	if params.centos6 {
+		commands = append(commands, `export VERSION=$(cat /go/.version.txt)`)
+		if !params.fips {
+			commands = append(commands,
+				`mv /go/artifacts/teleport-v$${VERSION}-linux-amd64-bin.tar.gz /go/artifacts/teleport-v$${VERSION}-linux-amd64-centos6-bin.tar.gz`,
+				`mv /go/artifacts/teleport-ent-v$${VERSION}-linux-amd64-bin.tar.gz /go/artifacts/teleport-ent-v$${VERSION}-linux-amd64-centos6-bin.tar.gz`,
+			)
+		} else {
+			commands = append(commands,
+				`mv /go/artifacts/teleport-ent-v$${VERSION}-linux-amd64-fips-bin.tar.gz /go/artifacts/teleport-ent-v$${VERSION}-linux-amd64-centos6-fips-bin.tar.gz`,
+			)
+		}
+	}
+
+	// generate checksums
+	commands = append(commands, fmt.Sprintf(`cd /go/artifacts && for FILE in teleport*%s; do sha256sum $FILE > $FILE.sha256; done && ls -l`, extension))
+	return commands
+}
+
+// tagPipelines builds all applicable tag pipeline combinations
+func tagPipelines() []pipeline {
+	var ps []pipeline
+	// regular tarball builds
+	for _, arch := range []string{"amd64", "386", "arm", "arm64"} {
+		for _, fips := range []bool{false, true} {
+			if (arch == "386" || arch == "arm") && fips {
+				// FIPS mode not supported on i386 or ARM
+				continue
+			}
+			ps = append(ps, tagPipeline(tagBuildType{os: "linux", arch: arch, fips: fips}))
+			// TODO(gus): support needs to be added upstream for building ARM/ARM64 packages first
+			// ps = append(ps, tagPackagePipeline(rpmPackage, tagBuildType{os: "linux", arch: arch, fips: fips}))
+			// ps = append(ps, tagPackagePipeline(debPackage, tagBuildType{os: "linux", arch: arch, fips: fips}))
+		}
+	}
+
+	// TODO(gus): needed until support is added upstream for building ARM/ARM64 packages
+	// Remove this section and uncomment above once this is added.
+	for _, arch := range []string{"amd64", "386"} {
+		for _, fips := range []bool{false, true} {
+			if (arch == "386" || arch == "arm") && fips {
+				// FIPS mode not supported on i386 or ARM
+				continue
+			}
+			ps = append(ps, tagPackagePipeline(rpmPackage, tagBuildType{os: "linux", arch: arch, fips: fips}))
+			ps = append(ps, tagPackagePipeline(debPackage, tagBuildType{os: "linux", arch: arch, fips: fips}))
+		}
+	}
+
+	// Only amd64 Windows is supported for now.
+	ps = append(ps, tagPipeline(tagBuildType{os: "windows", arch: "amd64"}))
+	// Also add the two CentOS 6 artifacts.
+	ps = append(ps, tagPipeline(tagBuildType{os: "linux", arch: "amd64", centos6: true}))
+	ps = append(ps, tagPipeline(tagBuildType{os: "linux", arch: "amd64", centos6: true, fips: true}))
+	return ps
+}
+
+// tagPipeline generates a tag pipeline for a given combination of os/arch/FIPS
+func tagPipeline(params tagBuildType) pipeline {
+	if params.os == "" {
+		panic("params.os must be set")
+	}
+	if params.arch == "" {
+		panic("params.arch must be set")
+	}
+
+	pipelineName := fmt.Sprintf("build-%s-%s", params.os, params.arch)
+	pipelineTitleSegment := "release artifacts"
```

Comment:
can this title be removed?
pipeline name should be clear enough on its own, steps can be just "build" and "copy"

Reply:
Yeah, this is basically a hangover from when I was validating the YAML by eye, I used it to make sure that I'd actually copied the correct block for RPM vs DEB vs tarball etc.